pkg/ocgorm/stats.go

```diff
@@ -118,31 +118,31 @@ var (
 		Name:        "go.sql/db/connections/wait_count",
 		Description: "The total number of connections waited for",
 		Measure:     MeasureWaitCount,
-		Aggregation: view.LastValue(),
+		Aggregation: view.Count(),
```

Comment:
Count() gives the number of metric measurement points that there are, is that what we are going for?

Reply:
Sorry, I am new, and I haven't pulled this repo. yet, I am going to do that, I think I am missing context.

Reply:
FYI, I was looking at this to try to grok what is going on, found it interesting:
https://github.com/go-gorm/prometheus/blob/master/stats.go

Reply:
One additional thing I found while trying to make sense of this that I found interesting:
https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1345

Looks like we have an ever-increasing value for wait count that should be reflected in the `LastValue`. Not to say this should not be a `Count`, it does make sense, but I am still confused about why the old way doesn't work.

Reply:
So for the gorm collection, it looks like we are registering and scheduling these OC views on an [interval](https://github.com/hashicorp/go-gin-gorm-opencensus/blob/master/pkg/ocgorm/stats.go#L167), and on that interval, we are building the DBStats object up each time. Given the definition of the OC count aggregation type The count of the number of measurement points., I linked it on the PR, but we do not create a measurement point each time a connection waits, this is something that is actually [set directly in the database]

(https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1345) object that golang/sql has to wrap the driver connection, and then is queried out when the DBStats object is [rendered](https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1203).

During these scheduled events, it seems like that would be when these measurement points would be created.
If we were emitting a metric each time a connection waited, then count would make perfect sense, but since we are relying on the database's internal reporting, I don't think it makes sense to use anything except the value that it reports.
Is there any way we could have a conversation about this to clear all this up? Sorry for the long message